crates/cli/src/package.rs

```diff
@@ -77,9 +78,52 @@ async fn internal_fetch(
     Ok(())
 }
 
+pub async fn install_single_package_to_virtual_store(
+    tarball_cache: &Cache,
+    http_client: &Client,
+    config: &'static Npmrc,
+    dependency_path: &DependencyPath,
+    package_snapshot: &PackageSnapshot,
+) -> Result<(), PackageManagerError> {
+    let PackageSnapshot { resolution, .. } = package_snapshot;
+    let LockfileResolution::Registry(registry_resolution) = resolution else {
+        panic!("Only TarballResolution is supported at the moment, but {dependency_path} requires {resolution:?}");
+    };
+
+    let DependencyPath { custom_registry, package_specifier } = dependency_path;
+    let registry = custom_registry.as_ref().unwrap_or(&config.registry);
+    let registry = registry.strip_suffix('/').unwrap_or(registry);
+    let PkgNameVerPeer { name, suffix: ver_peer } = package_specifier;
+    let version = ver_peer.version();
+    let bare_name = name.bare.as_str();
+    let tarball_url = format!("{registry}/{name}/-/{bare_name}-{version}.tgz");
+    let integrity = registry_resolution.integrity.as_str();
+
+    // TODO: skip when already exists in store?
+    let cas_paths = download_tarball_to_store(
+        tarball_cache,
+        http_client,
+        &config.store_dir,
+        integrity,
+        None,
+        &tarball_url,
+    )
```

Comment:
I suggest us to switch to using structs, when the number of arguments is more than 2.

Reply:
I will probably do a massive refactoring after this. So I doubt even the function `download_tarball_to_store` will be preserved.